cortex.tf

```diff
@@ -0,0 +1,144 @@
+locals {
+  cortex = merge(
+    local.helm_defaults,
+    {
+      name                      = "cortex"
+      namespace                 = "cortex"
+      chart                     = "cortex"
+      repository                = "https://cortexproject.github.io/cortex-helm-chart"
+      service_account_name      = "cortex"
+      create_iam_resources_irsa = true
+      enabled                   = false
+      chart_version             = "0.4.0"
+      version                   = "v1.7.0"
+      iam_policy_override       = ""
+      default_network_policy    = true
+      cluster_name              = "cluster"
+    },
+    var.cortex
+  )
+
+  values_cortex = <<VALUES
+nameOverride: "${local.cortex["name"]}"
+autoDiscovery:
+  clusterName: ${local.cortex["cluster_name"]}
+awsRegion: ${data.aws_region.current.name}
+rbac:
+  create: true
+  pspEnabled: true
+  serviceAccount:
+    annotations:
+      eks.amazonaws.com/role-arn: "${local.cortex["enabled"] && local.cortex["create_iam_resources_irsa"] ? module.iam_assumable_role_cortex[0].this_iam_role_arn : ""}"
+VALUES
+}
+
+module "iam_assumable_role_cortex" {
+  count                         = local.cortex["enabled"] && local.cortex["create_iam_resources_irsa"] ? 1 : 0
+  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version                       = "~> v2.0"
+  create_role                   = local.cortex["enabled"] && local.cortex["create_iam_resources_irsa"]
+  role_name                     = "tf-eks-${var.cluster-name}-cortex-irsa"
+  provider_url                  = replace(var.eks["cluster_oidc_issuer_url"], "https://", "")
+  role_policy_arns              = local.cortex["create_iam_resources_irsa"] ? [aws_iam_policy.eks-cortex[0].arn] : []
+  oidc_fully_qualified_subjects = ["system:serviceaccount:${local.cortex["namespace"]}:${local.cortex["service_account_name"]}"]
+}
+
+resource "aws_iam_policy" "eks-cortex" {
+  count  = local.cortex["enabled"] ? 1 : 0
+  name   = "tf-eks-${var.cluster-name}-cortex"
+  policy = local.cortex["iam_policy_override"] == "" ? data.aws_iam_policy_document.cortex[count.index].json : local.cortex["iam_policy_override"]
+}
+
+data "aws_iam_policy_document" "cortex" {
+  count = local.cortex["enabled"] ? 1 : 0
```

Comment:
could you also implement the `create_ns` logic like in the other addons for example [here](https://github.com/particuleio/terraform-kubernetes-addons/blob/main/karma.tf)